cloud_function/migrate_database.py

```
@@ -0,0 +1,586 @@
+        if table_exists(dest_conn, table_name):
+            return True
+
+        # Get table structure
+        source_cursor.execute(f"""
+            SELECT column_name, data_type, character_maximum_length,
+                   numeric_precision, numeric_scale, is_nullable, column_default
+            FROM information_schema.columns
+            WHERE table_schema = 'public'
+            AND table_name = '{table_name}'
+            ORDER BY ordinal_position;
+        """)
```

Ruff (0.14.1)

Possible SQL injection vector through string-based query construction

(S608)